docs/DevGuide/CompilerLinkerErrors.md

```diff
@@ -0,0 +1,99 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+
+# Understanding Compiler and Linker Errors {#compiler_and_linker_errors}
+
```

Comment:
This is a nice summary of the most common failure case for linker errors, but could you include a quick list at the top of the handful of things that are frequently missed that cause link problems -- the list isn't that long, I think. Probably include:
- missed a new `.cpp` file in a CMakeLists.txt
- absent explicit instantiation for a template function/class defined in a cpp
- intentionally undefined template object in an failing SFINAE branch
- missing library link (explained below)
That's at least the majority of my experience, and please add any more that you can think of from things you've encountered.